### PR TITLE
Updates json Tags for Defaulted API Fields

### PR DESCRIPTION
**Pull request by danehans**
Adds the `omitempty` json tag for defaulted API Fields. Previously, fields being defaulted did not contain `omitempty`. Without omitempty, an instance of `Contour` requires these fields which negates the desired defaulting behavior.

/assign @jpeach @stevesloka 
/cc @Miciah

### CODE REVIEW

File: api/v1alpha1/contour_types.go [hunk 1]
```diff
@@ -49,13 +49,13 @@ type ContourSpec struct {
 	//
 	// +kubebuilder:default=2
 	// +kubebuilder:validation:Minimum=0
-	Replicas int32 `json:"replicas"`
+	Replicas int32 `json:"replicas,omitempty"`
```
Comment (stevesloka) [inline api/v1alpha1/contour_types.go:52]: I thought this meant to be required from a previous different PR?

Comment (danehans) [reply api/v1alpha1/contour_types.go:52]: @stevesloka the field is required in a sense. It's not required for a user to set, but it's required when the object is persisted. For example, a user can create a contour object with:
```
apiVersion: operator.projectcontour.io/v1alpha1
kind: Contour
metadata:
  name: contour-sample
spec: {}
```

which will be realized (thanks for kubebuilder defaulting) to:
```
apiVersion: operator.projectcontour.io/v1alpha1
kind: Contour
metadata:
  name: contour-sample
spec:
  replicas: 2
  namespace:
    name: projectcontour
    removeOnDeletion: false
status:
<SNIP>
```

File: api/v1alpha1/contour_types.go [hunk 2]
```diff
@@ -49,13 +49,13 @@ type ContourSpec struct {
...
 
 	// namespace defines the schema of a Contour namespace.
 	// See each field for additional details.
 	//
-	// +optional
-	Namespace *NamespaceSpec `json:"namespace,omitempty"`
+	// +kubebuilder:default={name: "projectcontour", removeOnDeletion: false}
```
Comment (skriss) [inline api/v1alpha1/contour_types.go:57]: do we need to specify the defaults both here and on the fields within `NamespaceSpec`?

**Review by stevesloka — APPROVED** [review]
/lgtm

File: api/v1alpha1/contour_types.go (hunk 2 above)
Comment (danehans) [reply api/v1alpha1/contour_types.go:57]: @skriss it's not required, but I think it's helpful when reading the code or the CRD. I can remove in a follow-on PR if you prefer.

Comment (skriss) [reply api/v1alpha1/contour_types.go:57]: not a big deal, i think it can stay as-is.
